lib/ranked-model/ranker.rb

```diff
@@ -208,6 +214,10 @@ def rearrange_ranks
         end
       end
 
+      def handle_cannot_rearrange_ranks(message)
+        throw :ranking_failed, "Could not re-rank: #{message}."
```

Comment:
can be overridden to return nil to skip the re-ranking and not abort the save

default is to abort the save and add an error message